assembly/pallets/balances/assembly/balances-module.ts

```diff
@@ -0,0 +1,99 @@
+import { AccountData, AccountId, ISignedTransaction, TransactionValidity, ResponseCodes } from 'subsembly-core';
+import { Storage, Log} from "subsembly-core";
+import { ByteArray, BytesReader, UInt128, UInt64 } from "as-scale-codec";
+import { u128 } from "as-bignum";
+import { System } from "../../../frame";
+import { AccountIdType } from "../../../runtime/runtime";
+
+/**
+ * The Balances Module.
+ * Used for account balance manipulation such as:
+ *  - Getting and setting free/reserved balances
+ */
+export class Balances {
+
+    /**
+     * Returns AccountData for a given AccountId
+     * If the account does not exist, Default AccountData is returned.
+     */
+    static getAccountData(accountId: AccountIdType): AccountData {
+        const accDataBytes = Storage.get(accountId.getAddress());
+        if (accDataBytes.isSome()) {
+            return BytesReader.decodeInto<AccountData>((<ByteArray>accDataBytes.unwrap()).values);
+        } else {
+            return AccountData.getDefault();
+        }
+    }
+
+    /**
+     * Sets the balances of a given AccountId
+     * Alters the Free balance and Reserved balances in Storage.
+     */
+    static setBalance(accountId: AccountIdType, freeBalance: UInt128, reservedBalance: UInt128): AccountData {
+        const currentAccountData = this.getAccountData(accountId);
+
+        // TODO Any meaningful checks
+
+        currentAccountData.setFree(freeBalance);
+        currentAccountData.setReserved(reservedBalance);
+        Storage.set(accountId.getAddress(), currentAccountData.toU8a());
+
+        return currentAccountData;
+    }
+
+    /**
+     * Transfer the given amount from sender to receiver
+     * Note: this is just draft implementation, without necessary checks
+     * @param sender 
+     * @param receiver 
+     * @param amount 
+     */
+    static transfer(sender: AccountIdType, receiver: AccountIdType, amount: u64): void {
```

Comment:
I am wondering why on transfers the amount is `u64` but the balance is represented with `u128` in storage